collector/system_collector.go

```diff
@@ -296,8 +296,16 @@ func (s *SystemCollector) Collect(ch chan<- prometheus.Metric) {
 			} else if pcieDevices == nil {
 				systemLogContext.WithField("operation", "system.PCIeDevices()").Info("no PCI-E device data found")
 			} else {
+				processed := make(map[string]bool)
 				wg5.Add(len(pcieDevices))
 				for _, pcieDevice := range pcieDevices {
+					_, exists := processed[pcieDevice.ODataID]
+					if exists {
```

Comment:
nit: In the absence of any unit tests it would be helpful to add the scrape data where we see this issue to the `sampleOut` folder.

A comment could also be added here, referencing the sample data.

Reply:
Good idea. I have added sample data and comment about it.